middle_end/flambda/un_anf.ml

```diff
@@ -73,18 +73,46 @@ let closure_environment_var (ufunction:Clambda.ufunction) =
     (* closed function, no environment *)
     None
 
+type var_uses =
+  | Zero
+  | One
+  | More_than_one
+
+type var =
+  { definition_depth : int;
+    uses : var_uses; }
+
+let incr_uses { definition_depth; uses } depth =
+  assert (definition_depth <= depth);
+  let uses =
+    if definition_depth < depth then More_than_one
+    else begin
+      match uses with
+      | Zero -> One
+      | One -> More_than_one
+      | More_than_one -> More_than_one
+    end
+  in
+  { definition_depth; uses }
+
+let zero definition_depth = { definition_depth; uses = Zero }
+
+let add_definition t var depth =
+  V.Tbl.add t var (zero depth)
+
+let add_use t var depth =
+  match V.Tbl.find t var with
+  | info -> V.Tbl.replace t var (incr_uses info depth)
+  | exception Not_found -> ()
```

Comment:
Please add a comment explaining when the `Not_found` case is hit.

Reply:
Done